### DISCUSSION TITLE
Question about searchmoves

### DISCUSSION THREAD

**Discussion by dlbbld** [official-stockfish/Stockfish#3909] [Q&A] [answered]
I would like to understand how the "searchmoves" command and evaluation hang together.

First question:
For the below position run at depth 26, Stockfish gives f2f3 for the best move with cp 255.
rnb1kb1r/pppp1ppp/8/4q3/2B1n3/8/PPP1QPPP/RNB1K1NR w KQkq - 0 6

But now I would expect I get the same score if I run the below command, but I get cp 275:
go depth 26 searchmoves f2f3

Second question:
The above position after f2f3 is below:
rnb1kb1r/pppp1ppp/8/4q3/2B1n3/5P2/PPP1Q1PP/RNB1K1NR b KQkq - 0 6

At depth 25, Stockfish gives d7d5 for the best move with cp -240.

But I would expect that to match with "go depth 26 searchmoves f2f3" before the move as above with cp 272. So evaluating the move before playing with depth n should be the same as evaluating after the move with depth n-1.

 I started Stockfish separately for each example and ran only the commands to avoid caching affecting the score.

Can someone please explain how Stockfish works concerning these examples? Thanks.

**Comment by vondele** [comment, marked as answer]
searchmoves only restricts the list of moves searched at the root. What is being searched of course influences what is in the hash (transposition table), and the history heuristics that order the moves. As a result, search trees become very different, with slight differences in the search evaluation.

**Reply by dlbbld** [reply]
Thanks, that makes the behaviour clear.
